File: libs/reporting-api/src/lib/model/create-run-suite-payload.ts
import { CreateRunCasePayload } from './create-run-case-payload';
import { TestLocation } from './test-location';

export interface CreateRunSuitePayload {
  title: string | null;
  location?: TestLocation;
  runGroup?: string;
  suites?: CreateRunSuitePayload[];
  cases?: CreateRunCasePayload[];
}

File: backend/OpenWright.Api/Reporting/Payloads/CreateRunSuitePayload.cs
using OpenWright.Api.Projects.ValueObjects;

namespace OpenWright.Api.Reporting.Payloads;

public class CreateRunSuitePayload
{
    public required string? Title { get; init; }
    public TestLocation? Location { get; init; }
    public string? RunGroup { get; init; }
    public CreateRunSuitePayload[]? Suites { get; init; }
    public CreateRunCasePayload[]? Cases { get; init; }
}
